### PR TITLE
added view placeholders, routes and controllers

### PR DESCRIPTION
**Pull request by Samuelwanza**
I have conducted the following:
-For each URL I have created:

-A route.
-An action in the correct controller.
-A view file.
-View files include only placeholders written in plain HTML, e.g:
-Made sure that my added code to handle only the 4 required URLs.

### CODE REVIEW

File: spec/helpers/posts_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# describe PostsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe PostsHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (julie-ify) [inline spec/helpers/posts_helper_spec.rb]: - [ ] This file was generated with scaffold command. Kindly remove it.

File: spec/helpers/users_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# describe UsersHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe UsersHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (julie-ify) [inline spec/helpers/users_helper_spec.rb]: - [ ] This file was generated with scaffold command. Kindly remove it.

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb]: - [ ] This file was generated with scaffold command. Kindly remove it.

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (julie-ify) [inline spec/requests/users_spec.rb]: - [ ] This file was generated with scaffold command. Kindly remove it.

File: app/helpers/posts_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
+end
```
Comment (julie-ify) [inline app/helpers/posts_helper.rb]: - [ ] This file was generated with scaffold command. Kindly remove it.

File: app/helpers/users_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
Comment (julie-ify) [inline app/helpers/users_helper.rb]: - [ ] This file was generated with scaffold command. Kindly remove it.

File: config/routes.rb [hunk 7]
```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get "/users", to: "users#index", as: "users"
+  get "/users/:id", to: "users#show"
+  get "/users/:user_id/posts", to: "posts#index", as: "posts"
+  get "/users/:user_id/posts/:post_id", to: "posts#show"
```
Comment (julie-ify) [inline config/routes.rb:9]: - [ ] **[OPTIONAL]** You can define this better using `resources` adapter provided by Rails.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Samuelwanza,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Samuelwanza,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**